.github/workflows/coverage.yml

```diff
@@ -23,9 +22,8 @@ jobs:
         run: sudo -E make coverage
 
       - name: Upload coverage reports to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
-          fail_ci_if_error: true
+          fail_ci_if_error: false
```

Comment:
It might be good to retain this in order to watch for failures unrelated to the token error. As far as I can see, it shouldn't affect the job status for tokenless upload if everything goes right.